### PR TITLE
would like to set retention period for specific topic #27

### PR DESCRIPTION
**Pull request by adison88**
* Added Topic level setting for retention period.

### CODE REVIEW

**Comment by adison88** [comment]
@jonathansant please review.

**Comment by AmrutaMoghe** [comment]
@adison88 
Please update the readme file as well.

**Comment by jonathansant** [comment]
Thank you for your PR. I will review shortly. In the meantime can you please update the version in the `package.json` file

File: .gitignore [hunk 1]
```diff
@@ -262,3 +262,4 @@ __pycache__/
 
 # project items
 /stack.local.yml
+
```
Comment (jonathansant) [inline .gitignore]: remove

File: Orleans.Streams.Kafka/Config/KafkaStreamOptions.cs [hunk 2]
```diff
@@ -120,6 +121,14 @@ public class TopicConfig
...
 		/// <remarks>1 by default</remarks>
 		public short ReplicationFactor { get; set; } = 1;
+
+		/// <summary>
+		/// If <see cref="RetentionPeriodInMs"/> is set the topic will
+		/// be created with only retain data for this much duration.
+		/// If not set, it'll take default configuration of broker which is 7 days.
+		/// </summary>
+		/// <remarks>7 days by default</remarks>
+		public ulong RetentionPeriodInMs { get; set; } = 604800000;
```
Comment (jonathansant) [inline Orleans.Streams.Kafka/Config/KafkaStreamOptions.cs]: can you make this `nullable`? as it is any AutoCreated Topic will have a default retention period of 7 days. I would prefer if there is **_no_** a retention period by default

File: .gitignore (hunk 1 above)
Comment (adison88) [reply .gitignore]: done.

File: Orleans.Streams.Kafka/Config/KafkaStreamOptions.cs (hunk 2 above)
Comment (adison88) [reply Orleans.Streams.Kafka/Config/KafkaStreamOptions.cs]: Updated.

**Comment by adison88** [comment]
@jonathansant Addressed the comments.

**Comment by jonathansant** [comment]
Great! Can you please update the changelog so we can merge 😄

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "orleans.streams.kafka",
-  "version": "3.0.0",
+  "version": "3.0.1",
```
Comment (jonathansant) [inline package.json]: can you bump the minor, since it's a new feature? Thanks.

Comment (adison88) [reply package.json]: Bumped up the minor version and updated the changelog. @jonathansant

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,10 @@
-﻿## [3.0.0](https://github.com/jonathansant/orleans.streams.kafka/compare/2.0.1...3.0.0) (2019-10-25)
+﻿## [3.1.0] (2020-03-11)
```
Comment (jonathansant) [inline CHANGELOG.md]:
```suggestion
## [3.1.0](https://github.com/jonathansant/orleans.streams.kafka/compare/3.0.0...3.1.0) (2020-03-12)
```

**Comment by adison88** [comment]
@jonathansant suggestion applied.

**Pull request merged**